core/src/main/java/org/elasticsearch/rest/action/document/RestBulkAction.java

```diff
@@ -86,8 +86,10 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
         bulkRequest.timeout(request.paramAsTime("timeout", BulkShardRequest.DEFAULT_TIMEOUT));
         bulkRequest.setRefreshPolicy(request.param("refresh"));
-        bulkRequest.add(request.content(), defaultIndex, defaultType, defaultRouting, defaultFields,
-            defaultFetchSourceContext, defaultPipeline, null, allowExplicitIndex, request.getXContentType());
+        if(request.hasContent()) {
+            bulkRequest.add(request.content(), defaultIndex, defaultType, defaultRouting, defaultFields,
+                defaultFetchSourceContext, defaultPipeline, null, allowExplicitIndex, request.getXContentType());
+        }
 
```

Comment:
I followed the pattern used in a few other places. e.g. [RestCreateIndexAction#L44-L46](https://github.com/elastic/elasticsearch/blob/7420bda8ed8a992b36ec6bd1b8b8c11b8b20db44/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java#L44-L46).
Instead of failing in `RestHandler.prepareRequest` a validation exception will be thrown from `ActionRequest.validate()` if request content is missing but required.
Would an approach similar to [RestDeleteByQueryAction.java#L51-L53](https://github.com/elastic/elasticsearch/blob/2e48fb8294a924bc3742419fe80e4a29fdfa0f54/modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestDeleteByQueryAction.java#L51-L53) be better for cases when request content is required?